openstf_dbc/data_interface.py

```diff
@@ -40,55 +57,50 @@ def __init__(self):
         # such as macOS or Windows native preferences – see
         # urllib.request.ProxyHandler for more details).
         # The proxies value for using system proxies is None.
-        geopy.geocoders.options.default_proxies = self.config.proxies
+        geopy.geocoders.options.default_proxies = config.proxies
 
-        self._connect_to_database(use_influxdb=True, use_mysql=True)
+        _DataInterface._instance = self
 
     @staticmethod
     def get_instance():
-        if _DataInterface._instance is None:
-            _DataInterface._instance = _DataInterface()
-        return _DataInterface._instance
-
-    def _connect_to_database(self, use_influxdb=True, use_mysql=True):
-        # Create Influx client
-        if use_influxdb is True:
-            self.influx_client = self._create_influx_client()
-
-        # Create SQL engine
-        if use_mysql is True:
-            self.sql_engine = self._create_sql_write_engine()
-
-    def _create_influx_client(self):
+        try:
+            return Singleton.get_instance(_DataInterface)
+        except KeyError as exc:
+            # if _DataInterface not in Singleton._instances:
+            raise RuntimeError(
+                "No _DataInterface instance initialized. "
+                "Please call _DataInterface(config) first."
+            ) from exc
+        # return Singleton._instances[_DataInterface]
+
```

Comment:
Line 69 and 74 look like commented out code

Reply:
done